rosbag2_transport/test/rosbag2_transport/test_play.cpp

```diff
@@ -174,6 +174,68 @@ TEST_F(RosBag2PlayTestFixture, recorded_messages_are_played_for_all_topics)
           ElementsAre(40.0f, 2.0f, 0.0f)))));
 }
 
+TEST_F(RosBag2PlayTestFixture, recorded_messages_are_played_for_all_topics_multi)
+{
+  auto msg = get_messages_basic_types()[0];
+  msg->int32_value = 42;
+
+  auto topic_types = std::vector<rosbag2_storage::TopicMetadata>{
+    {1u, "topic1", "test_msgs/BasicTypes", "", {}, ""},
+    {2u, "topic2", "test_msgs/BasicTypes", "", {}, ""},
```

Comment:
Need to avoid ambiguity in message types. It could be a services yet.
```suggestion
    {1u, "topic1", "test_msgs/msg/BasicTypes", "", {}, ""},
    {2u, "topic2", "test_msgs/msg/BasicTypes", "", {}, ""},
```

Reply:
changed in fd349cc793eab28feb6ef089c20ad1a6329a4001

should this be done for all other tests then? Maybe in a separate PR?

Reply:
Yes. we do this for all other tests in Apex.AI and then going to upstream changes.